DonoGear/app/src/main/java/com/example/donogear/actionpages/SearchPageFragment.java

```diff
@@ -169,13 +171,14 @@ private void displayData() {
      * @param position - position of the clicked item in the list
      */
     @Override
-    public void onItemClick(View view, int position) {
-        ItemDetails item = listOfItems.get(position);
-//        System.out.println(item.itemName);
-        Intent intent = new Intent(activity, ProductDetails.class);
-        intent.putExtra("item_details", item);
-        intent.putExtra("typeOfSearch", typeOfSearch);
-        startActivity(intent);
+    public void onItemClick(View view, int position, String type) {
+        if (type.equals(SEARCH)) {
+            ItemDetails item = listOfItems.get(position);
+            Intent intent = new Intent(activity, ProductDetails.class);
+            intent.putExtra("item_details", item);
```

Comment:
Should create variables for item_details and typeOfSearch. Since they are used in multiple files it might make sense to add them to the Constants file